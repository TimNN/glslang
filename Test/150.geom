#version 150 core

in fromVertex {
    vec3 color;
} fromV;

out toFragment {
    vec3 color;
} toF;

out fromVertex {  // okay to reuse a block name for another block name
    vec3 color;
};

out fooB {
    vec2 color;
} fromVertex;     // ERROR, cannot reuse block name as block instance

int fromVertex;   // ERROR, cannot reuse a block name for something else

out fooC {
    vec2 color;
} fooC;           // ERROR, cannot have same name for block and instance name

void main()
{
    EmitVertex();
    EndPrimitive();
    EmitStreamVertex(1);    // ERROR
    EndStreamPrimitive(0);  // ERROR

    color = fromV.color;
    gl_ClipDistance[3] = gl_in[1].gl_ClipDistance[2];
    gl_Position = gl_in[0].gl_Position;
    gl_PointSize = gl_in[3].gl_PointSize;
    gl_PrimitiveID = gl_PrimitiveIDIn;
    gl_Layer = 2;
}
