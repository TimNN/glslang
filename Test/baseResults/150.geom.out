Warning, version 150 is not yet complete; some version-specific features are present, but many are missing.
ERROR: 0:15: 'fromVertex' : block instance name redefinition 
ERROR: 0:19: 'fromVertex' : redefinition 
ERROR: 0:21: 'fooC' : block instance name redefinition 
ERROR: 0:29: 'EmitStreamVertex' : no matching overloaded function found 
ERROR: 0:30: 'EndStreamPrimitive' : no matching overloaded function found 
ERROR: 5 compilation errors.  No code generated.

ERROR: node is still EOpNull!
0:25  Function Definition: main( (void)
0:25    Function Parameters: 
0:27    Sequence
0:27      EmitVertex (void)
0:28      EndPrimitive (void)
0:29      Constant:
0:29        0.000000
0:30      Constant:
0:30        0.000000
0:32      move second child to first child (3-component vector of float)
0:32        color: direct index for structure (3-component vector of float)
0:32          '__anon__0' (out block)
0:32          Constant:
0:32            0 (const uint)
0:32        color: direct index for structure (3-component vector of float)
0:32          'fromV' (in block)
0:32          Constant:
0:32            0 (const int)
0:33      move second child to first child (float)
0:33        direct index (float)
0:33          gl_ClipDistance: direct index for structure (unsized array of float)
0:33            '__anon__1' (out block)
0:33            Constant:
0:33              2 (const uint)
0:33          Constant:
0:33            3 (const int)
0:33        direct index (float)
0:33          gl_ClipDistance: direct index for structure (unsized array of float)
0:33            direct index (in block)
0:33              'gl_in' (in unsized array of block)
0:33              Constant:
0:33                1 (const int)
0:33            Constant:
0:33              2 (const int)
0:33          Constant:
0:33            2 (const int)
0:34      move second child to first child (4-component vector of float)
0:34        gl_Position: direct index for structure (4-component vector of float)
0:34          '__anon__1' (out block)
0:34          Constant:
0:34            0 (const uint)
0:34        gl_Position: direct index for structure (4-component vector of float)
0:34          direct index (in block)
0:34            'gl_in' (in unsized array of block)
0:34            Constant:
0:34              0 (const int)
0:34          Constant:
0:34            0 (const int)
0:35      move second child to first child (float)
0:35        gl_PointSize: direct index for structure (float)
0:35          '__anon__1' (out block)
0:35          Constant:
0:35            1 (const uint)
0:35        gl_PointSize: direct index for structure (float)
0:35          direct index (in block)
0:35            'gl_in' (in unsized array of block)
0:35            Constant:
0:35              3 (const int)
0:35          Constant:
0:35            1 (const int)
0:36      move second child to first child (int)
0:36        'gl_PrimitiveID' (out int)
0:36        'gl_PrimitiveIDIn' (in int)
0:37      move second child to first child (int)
0:37        'gl_Layer' (out int)
0:37        Constant:
0:37          2 (const int)
0:?   Linker Objects
0:?     'fromV' (in block)
0:?     'toF' (out block)
0:?     '__anon__0' (out block)

